### PR TITLE
*: add extractor for cluster_tidb_index_usage

### PR DESCRIPTION
**Pull request by joechenrh**
### What problem does this PR solve?

Issue Number: ref #50305

### What changed and how does it work?

`InfoSchemaIndexUsageExtractor` is added for cluster_tidb_index usage memtable.

Besides, the explain output of cluster memory table is changed to make it consistent with other memtable.

```
mysql> explain select * from information_schema.CLUSTER_TIDB_INDEX_USAGE where table_schema = "test";    
+------------------------+----------+-----------+--------------------------------+----------------------------------------------------------------------+
| id                     | estRows  | task      | access object                  | operator info                                                        |
+------------------------+----------+-----------+--------------------------------+----------------------------------------------------------------------+
| TableReader_7          | 10.00    | root      |                                | data:Selection_6                                                     |
| └─Selection_6          | 10.00    | cop[tidb] |                                | eq(information_schema.cluster_tidb_index_usage.table_schema, "test") |
|   └─MemTableScan_5     | 10000.00 | cop[tidb] | table:CLUSTER_TIDB_INDEX_USAGE |                                                                      |
+------------------------+----------+-----------+--------------------------------+----------------------------------------------------------------------+
```

Before:

```
mysql> explain select * from information_schema.CLUSTER_TIDB_INDEX_USAGE where table_schema = "test";                                                                                 
+-------------------------+----------+-----------+--------------------------------+----------------------------------------------------------------------+                            
| id                      | estRows  | task      | access object                  | operator info                                                        |                            
+-------------------------+----------+-----------+--------------------------------+----------------------------------------------------------------------+                            
| TableReader_7           | 10.00    | root      |                                | data:Selection_6                                                     |                            
| └─Selection_6           | 10.00    | cop[tidb] |                                | eq(information_schema.cluster_tidb_index_usage.table_schema, "test") |                            
|   └─TableFullScan_5     | 10000.00 | cop[tidb] | table:CLUSTER_TIDB_INDEX_USAGE | keep order:false, stats:pseudo                                       |                            
+-------------------------+----------+-----------+--------------------------------+----------------------------------------------------------------------+                            
```

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/executor/infoschema_reader.go [hunk 1]
```diff
@@ -3691,7 +3691,7 @@ func (e *memtableRetriever) setDataForClusterIndexUsage(ctx context.Context, sct
 	dom := domain.GetDomain(sctx)
 	rows := make([][]types.Datum, 0, 100)
 	checker := privilege.GetPrivilegeManager(sctx)
-	extractor, ok := e.extractor.(*plannercore.InfoSchemaBaseExtractor)
+	extractor, ok := e.extractor.(*plannercore.InfoSchemaIndexUsageExtractor)
```
Comment (tangenta) [inline pkg/executor/infoschema_reader.go:3694]: Maybe add a method `ListSchemasAndTables` for `InfoSchemaIndexUsageExtractor` and avoid using `SchemaTableInfos`.

Comment (joechenrh) [reply pkg/executor/infoschema_reader.go:3694]: `InfoSchemaIndexUsageExtractor` has already implemented schema and table filter, which is used in tidb_index_usage.
The only difference is that for cluster_tidb_index_usage, the request needs to be sent to all tidb instances for execution

File: pkg/executor/infoschema_reader.go [hunk 2]
```diff
@@ -3801,9 +3741,8 @@ func (e *memtableRetriever) setDataFromIndexUsage(ctx context.Context, sctx sess
 	return nil
 }
 
-// Currently, ClusterIndexUsage will not be be pushed down, so just use the previous logic
-func (e *memtableRetriever) setDataFromClusterIndexUsage(ctx context.Context, sctx sessionctx.Context, schemas []model.CIStr) error {
-	err := e.setDataForClusterIndexUsage(ctx, sctx, schemas)
+func (e *memtableRetriever) setDataFromClusterIndexUsage(ctx context.Context, sctx sessionctx.Context) error {
+	err := e.setDataFromIndexUsage(ctx, sctx)
```
Comment (tiancaiamao) [inline pkg/executor/infoschema_reader.go:3745]: What's the difference between clustered_index_usage table and tidb_index_usage table?

**Review by tiancaiamao — APPROVED** [review]

File: pkg/executor/infoschema_reader.go (hunk 2 above)
Comment (tiancaiamao) [reply pkg/executor/infoschema_reader.go:3745]: OK, different tidb at the same time may see different schema.

Comment (joechenrh) [reply pkg/executor/infoschema_reader.go:3745]: https://docs.pingcap.com/tidb/stable/information-schema-tidb-index-usage#tidb_index_usage

I think each tidb will collect its own index-related statistics.

Comment (joechenrh) [reply pkg/executor/infoschema_reader.go:3745]: For example, I ran some query on one node and got this output.
```
mysql> select * from information_schema.CLUSTER_TIDB_INDEX_USAGE where table_schema = "test" and index_name = "idx_1";
+-----------------+--------------+------------+------------+-------------+--------------+-------------------+---------------------+-----------------------+------------------------+-------------------------+-------------------------+--------------------------+-----------------------+---------------------+
| INSTANCE        | TABLE_SCHEMA | TABLE_NAME | INDEX_NAME | QUERY_TOTAL | KV_REQ_TOTAL | ROWS_ACCESS_TOTAL | PERCENTAGE_ACCESS_0 | PERCENTAGE_ACCESS_0_1 | PERCENTAGE_ACCESS_1_10 | PERCENTAGE_ACCESS_10_20 | PERCENTAGE_ACCESS_20_50 | PERCENTAGE_ACCESS_50_100 | PERCENTAGE_ACCESS_100 | LAST_ACCESS_TIME    |
+-----------------+--------------+------------+------------+-------------+--------------+-------------------+---------------------+-----------------------+------------------------+-------------------------+-------------------------+--------------------------+-----------------------+---------------------+
| 127.0.0.1:10081 | test         | t          | idx_1      |           0 |            0 |                 0 |                   0 |                     0 |                      0 |                       0 |                       0 |                        0 |                     0 | NULL                |
| 127.0.0.1:10080 | test         | t          | idx_1      |           1 |            1 |                 3 |                   0 |                     0 |                      0 |                       0 |                       0 |                        1 |                     0 | 2024-08-15 09:46:04 |
+-----------------+--------------+------------+------------+-------------+--------------+-------------------+---------------------+-----------------------+------------------------+-------------------------+-------------------------+--------------------------+-----------------------+---------------------+
```

**Review by tangenta — APPROVED** [review]

**Review by GMHDBJD — APPROVED** [review]

**Review by hawkingrei — APPROVED** [review]

**Pull request merged**